scripts/yse/build_parent_sample.py

```diff
@@ -0,0 +1,191 @@
+    keys_data.remove('FLT')
+
+
+    for i in range(num_examples):
+        # For this example, find the band with the most observations
+        # and store the number of observations as max_length
+        _, count = np.unique(data['FLT'][i], return_counts=True)
+        max_length = count.max()
+
+        # Create mask to select data from each timeseries by band
+        mask = np.expand_dims(all_bands, 1) == data['FLT'][i]
+
+        for key in keys_data:
+            timeseries_all_bands = []  # Stores a particular timeseries (as specified by the key) in all bands
+            for j in range(len(all_bands)):
+                timeseries_band = data[key][i][mask[j]]  # Select samples from timeseries for a particular band
+                timeseries_band = np.pad(  # Pad single band timeseries to max_length
+                    timeseries_band,
+                    (0, max_length - len(timeseries_band)),
+                    mode='constant',
+                    constant_values=-99 if key == 'MJD' else 0
+                )
+                timeseries_all_bands.append(timeseries_band)
+            timeseries_all_bands = convert_dtype(np.array(timeseries_all_bands))
+            data[key][i] = timeseries_all_bands
+
+    # Convert metadata to numpy arrays and cast to required datatypes
+    for key in keys_metadata:
+        metadata[key] = convert_dtype(np.array(metadata[key]))
+    
+    # Add numeric object_id to metadata (integer for each example in order of reading files)
+    keys_metadata.append('object_id')
+    metadata['object_id'] = np.arange(1, num_examples + 1)
+
+    # Add healpix to metadata
+    keys_metadata.append('healpix')
+    metadata['healpix'] = hp.ang2pix(16, metadata['RA'], metadata['DECL'], lonlat=True, nest=True)
+
+    # Cast bands to required datatype
+    all_bands = convert_dtype(all_bands)
+
+    # Establish conversions to standard names
+    keys_all = keys_metadata + keys_data
+    name_conversion = dict(zip(keys_all, keys_all))
+    name_conversion.update({
+        'RA': 'ra',
+        'DECL': 'dec',
+        'MJD': 'time',
+        'FLUXCAL': 'flux',
+        'FLUXCALERR': 'flux_err',
+        'HOST_LOGMASS': 'host_log_mass',
+        'SPEC_CLASS': 'spec_class',
+    })
+    # map 'redshift' depending on which keys are available
+    key_options_list = {
+        'redshift': ['REDSHIFT_FINAL', 'REDSHIFT_CMB', 'REDSHIFT_HELIO'],
+        'host_log_mass': ['HOST_LOGMASS', 'HOSTGAL_LOGMASS']
+    }
+
+    # Make output directories labelled by healpix
+    unique_healpix = np.unique(metadata['healpix'])
+    healpix_num_digits = len(str(hp.nside2npix(16)))
+    for healpix in unique_healpix:
+        healpix = str(healpix).zfill(healpix_num_digits)
+        os.makedirs(os.path.join(args.output_dir, f'healpix={healpix}'), exist_ok=True)
+
+    # Save data as hdf5 grouped into directories by healpix
+    object_id_num_digits = len(str(num_examples))
+    for i in range(num_examples):
```

Comment:
The image people do this as far as I know. Requesting comments from @benboyd97 @tom-hehir.

Reply:
YSE is a relatively small data set, there are 1975 SNe

Reply:
As discussed today, it would probably be okay to leave as individual files for version 1, but either convention could make sense so we could review this at some point.

For future reference, we did previously store a single hdf5 file per healpix. So if we want to change this later, we should be able to reuse code from previous commits.

Reply:
Keeping files in this format.